base/linalg/transpose.jl

```diff
@@ -0,0 +1,334 @@
+immutable Transpose{T,S<:AbstractMatrix} <: AbstractMatrix{T}
+    data::S
+    conjugated::Bool
+end
```

Comment:
I agree, the non-conjugated transpose is often useful.

I actually think we need a separate `Conj` wrapper. It makes the conjugation and transposition a bit more modular. You could use a `MappedArray` for `Conj` but it needs to be easy for dispatch on `*(::Transpose{Conj{Matrix}}, ::Matrix)` to be equivalent to what `Ac_mul_B` is now.

But in either case I don't think having it here as a field parameter is useful design since conjugation is normally determined at compile time (for example, it would be a minor performance hitch in *StaticArrays.jl*).